compiler/rustc_trait_selection/src/traits/object_safety.rs

```diff
@@ -112,33 +112,35 @@ fn object_safety_violations_for_trait(
                             tcx.def_path_str(trait_def_id)
                         ));
                         let node = tcx.hir().get_if_local(trait_def_id);
-                        let msg = if let Some(hir::Node::Item(item)) = node {
-                            err.span_label(
+                        let mut spans = MultiSpan::from_span(*span);
```

Comment:
can you move this entire `struct_span_lint_hir` invocation into a separate function (since you're already modifying all of it, that should not cause further churn I think)